metals/src/main/scala/scala/meta/internal/bsp/BspConnector.scala

```diff
@@ -62,6 +62,7 @@ class BspConnector(
           scribe.info("No build server found")
           Future.successful(None)
         case ResolvedBloop =>
+          scribe.info(s"Start new Bloop server on port ${userConfiguration.bloopNailgunPort.getOrElse("default port")}")
```

Comment:
So I'm not actually 100% sure what happens with Bloop if the server is already running with the default port, and then Metals is started with this setting. @tgodzik how does Bloop respond here? 

- Does it just detect that Bloop is already running and therefore go right to the BSP connection like it normally would?
- Or does the launcher detect that a new port is desired, and then shutdown the other and start a new one?

Reply:
Also related, depending on what happens in the above scenario, we may want to change this logging a bit to only log in the case that the port is defined, since falling into this case doesn't always mean that a _new_ server will be started.

Reply:
> So I'm not actually 100% sure what happens with Bloop if the server is already running with the default port, and then Metals is started with this setting. @tgodzik how does Bloop respond here?
> 
>     * Does it just detect that Bloop is already running and therefore go right to the BSP connection like it normally would?
> 
>     * Or does the launcher detect that a new port is desired, and then shutdown the other and start a new one?

As I've understood, Metals can't detect Bloop on any different port. It just starts new Bloop on new port. We'll have 2 running Bloop.
Only if we start Bloop on default port, change IDE metals properties and restart Metals with IDE. In this case we'll have 1 running Bloop on new port.

Reply:
> As I've understood, Metals can't detect Bloop on any different port. It just starts new Bloop on new port. We'll have 2 running Bloop.

I don't believe this is correct as there can only be one Bloop instance running at a time on a machine.

Reply:
I am not sure how it's synchronized, but this might indeed be two Bloop server, I will check how it works in this PR to make sure.

Reply:
> but this might indeed be two Bloop server

Really? I was always under the assumption that this was a limitation of Bloop since right in the docs it says

> There can only be one bloop server instance running per machine

https://scalacenter.github.io/bloop/docs/what-is-bloop#limitations

Reply:
You might be right, I am currently just wondering if we ever enforce it.

Reply:
Nobody could prohibit me run several bloop servers:
`bloop server`
`bloop server 8214`
`bloop server 8215`
I've done it right now :)

It's another thing bloop cli can't detect server on port different from default
`bloop about` `bloop project` always start server on default port